### DISCUSSION TITLE
Fixing incorrect resource specifications

### DISCUSSION THREAD

**Discussion by wkharold** [flux-framework/flux-core#4420] [General] [answered]
When you make a mistake with `flux R encode`, e.g., get --cores wrong, how do you correct it?

I did just that and tried to fix it by replacing `system/R` and restarting flux but it doesn't mark the previously misconfigured nodes as up.

`flux resource drain` shows the 'missing resources: core ...' error on those nodes. Restarting everything, manager node, compute nodes, etc. doesn't correct it so I assume there's something saved in `kvs` that needs to be corrected.

**Comment by garlick** [comment, marked as answer]
Have you tried flux resource undrain?   Nodes stay drained across a restart.
